src/twisted/internet/interfaces.py

```diff
@@ -189,10 +190,10 @@ class IHostnameResolver(Interface):
     def resolveHostName(
         resolutionReceiver: IResolutionReceiver,
         hostName: str,
-        portNumber: int,
-        addressTypes: Sequence[IAddress],
-        transportSemantics: str,
-    ) -> IResolutionReceiver:
+        portNumber: int = 0,
+        addressTypes: Optional[Sequence[Type[IAddress]]] = None,
+        transportSemantics: str = "TCP",
+    ) -> IHostResolution:
```

Comment:
changing interfaces is complicated ... I think the only way to change an interface, is to add a new one.

Reply:
I'm not changing the return value of the interface -- the type hints were incorrect when they were added.